src/qililab/instruments/agilent/e5071b_vna.py

```diff
@@ -51,13 +52,15 @@ def electrical_delay(self, time: float):
             value (str) : Electrical delay in ns
         """
         self.settings.electrical_delay = time
-        self.send_command("CALC:MEAS:CORR:EDEL:TIME", f"{time}")
+        if hasattr(self, "device") and self.device is not None:
```

Comment:
Since this check is repeated everywhere you should create a method that includes it, for example `is_device_initialized()`. It can be in the `Instrument` class level, since `device` is common for all instruments.

Reply:
That's a great idea!